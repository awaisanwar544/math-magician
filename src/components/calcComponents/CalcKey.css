.calc-btn {
  background-color: #d3d3d3;
  border: none;
  color: #383838;
  padding: 1em 2em;
  text-decoration: none;
  cursor: pointer;
  font-weight: 600;
  font-size: 1em;
  border: solid 1px #383838;
}

.operator {
  background-color: #ffa500;
}