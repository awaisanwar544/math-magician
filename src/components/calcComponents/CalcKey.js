import PropTypes from 'prop-types';
import './CalcKey.css';

const CalcKey = (props) => {
  const { type, keyValue } = props;
  const classNames = `calc-btn ${type}`;
  return (
    <input className={classNames} value={keyValue} type="button" />
  );
};

CalcKey.propTypes = {
  type: PropTypes.string.isRequired,
  keyValue: PropTypes.string.isRequired,
};

export default CalcKey;
