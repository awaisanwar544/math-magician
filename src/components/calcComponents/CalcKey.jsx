import PropTypes from 'prop-types';
import './CalcKey.css';

const CalcKey = (props) => {
  const { type, keyValue, onClick } = props;
  const classNames = `calc-btn ${type}`;
  return (
    <input
      className={classNames}
      value={keyValue}
      type="button"
      onClick={onClick}
    />
  );
};

CalcKey.propTypes = {
  type: PropTypes.string.isRequired,
  keyValue: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default CalcKey;
