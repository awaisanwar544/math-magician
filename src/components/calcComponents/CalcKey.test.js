import React from 'react';
import renderer from 'react-test-renderer';

import CalcKey from './CalcKey';

describe('Calculator Keys', () => {
  it('Displays correct text on the calculator key', () => {
    const calculatorKey = renderer
      .create(<CalcKey
        key={7}
        category="other"
        keyValue="7"
        onClick={jest.fn(() => 7)}
      />)
      .toJSON();
    expect(calculatorKey).toMatchSnapshot();
  });
});
