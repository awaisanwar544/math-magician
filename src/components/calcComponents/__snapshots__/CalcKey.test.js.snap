// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator Keys Displays correct text on the calculator key 1`] = `
<input
  className="calc-btn other"
  data-testid="calc-key"
  onClick={[MockFunction]}
  type="button"
  value="7"
/>
`;
